build.gradle

```diff
@@ -80,20 +84,80 @@ project(":service") {
     shadowJar {
         archiveName = "commercetools-payone-integration.jar"
         manifest {
-            attributes 'Main-Class': 'com.commercetools.pspadapter.payone.ServiceFactory'
+            attributes 'Main-Class': project.mainClassName
+
+            // set version to define at runtime using getImplementationVersion()
+            attributes 'Implementation-Title': rootProject.name
+            attributes 'Implementation-Version': allprojects.version
+        }
+
+        doLast {
+            println "Packaged JAR to ${destinationDir}/${archiveName}"
+        }
+    }
+
+    run {
+        dependsOn ':service:convertGradlePropertiesToJvmArgs'
+    }
+
+    task convertGradlePropertiesToJvmArgs (type:Task) {
+        description 'Convert the gradle build properties (set in ~/.gradle/gradle.properties) ' +
+                    'to JVM runtime properties (-Dkey=value arguments)'
+
+        doLast {
+            applicationDefaultJvmArgs = assertAndGetServiceProperties(project)
         }
     }
 }
 
 /**
+ * Convert the list of required properties from gradle build properties or environment variables
+ * to java runtime properties (supplied as <b>-Dkey=value</b> arguments).
+ * <p>
+ * The function tries to find a property value from the {@code project} properties and then from environment variables.
+ * If neither of them exists - throw an exception (fail-fast)
+ * <p>
+ *     Opposite to {@link #assertExecutableSpecRequirementsForTests(org.gradle.api.Project, org.gradle.api.tasks.testing.Test)}
+ *     this function is used for the service runtime.
+ *
+ * @param project gradle project from which to take the properties
+ * @return a list of java runtime argument in <i>-Dkey=value</i> syntax.
+ *
+ * @throws GradleException if any required system property or environment variable is undefined or empty
+ * @see #assertExecutableSpecRequirementsForTests(org.gradle.api.Project, org.gradle.api.tasks.testing.Test)
+ */
+static List<String> assertAndGetServiceProperties(Project project) {
```

Comment:
what is the use case to read arguments like ```-Dkey=value``` and not directly from env vars. Also why this validation is not executed in the service which requires them but here during build process?